docker/Dockerfile

```diff
@@ -52,6 +52,8 @@ FROM rosdep-depend AS rosdep-universe-localization-mapping-depend
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ARG ROS_DISTRO
 
+COPY src/universe/autoware.universe/launch/tier4_localization_launch /autoware/src/universe/autoware.universe/launch/tier4_localization_launch
```

Comment:
tier4_localization_launch depends on the following packages from other folders:
* autoware.universe/system/default_ad_api_helpers/automatic_pose_initializer
* autoware.universe/sensing/autoware_pcl_extensions
* autoware.universe/sensing/autoware_pointcloud_preprocessor 

Should we copy these packages as well?

Reply:
If `exec_depend` is properly written, the `health-check` should fail when there are missing dependency packages, but it is succeeding.
https://github.com/autowarefoundation/autoware/actions/runs/12413593055/job/34655835144?pr=5586

It seems necessary to actually create the image and test whether it launches with `ros2 launch`.

Reply:
I created container images with all PRs merged in the forked repository.
https://github.com/youtalk/autoware/pull/158
https://hub.docker.com/repository/docker/youtalk/autoware/tags?name=20241220

`ros2 launch` has started, but I don’t know what parameters to provide. @mitsudome-r Could you check the launch validity instead of me?

```
docker run --rm -it youtalk/autoware:universe-localization-mapping-20241220-amd64 ros2 launch tier4_localization_launch localization.launch.xml
[INFO] [launch]: All log files can be found below /root/.ros/log/2024-12-20-05-47-32-395432-b78797c5f174-1
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Included launch description missing required argument 'pose_source' (description: 'A string consisting of ndt, yabloc, artag and eagleye joined by underscores no matter the order. e.g. ndt_yabloc'), given: []
```

Reply:
> If exec_depend is properly written, the health-check should fail when there are missing dependency packages, but it is succeeding.

The rosdep script is actually outputting error messages, but the build passes because they are just exec depends.
We might need to report it as a CI error when ever we get error messages out of the scripts.
![image](https://github.com/user-attachments/assets/bbbadb31-bdae-4943-8d89-97372d4a98bf)

> ros2 launch has started, but I don’t know what parameters to provide. @mitsudome-r Could you check the launch validity instead of me?

At the moment, the launch file is expected to be launched via autoware.launch.xml so it requires lengthy arguments. I will write them down here. Give me few minutes.

Reply:
In order to launch you need to give arguments and parameters to the launch file:
1. Download [localization_config.zip](https://github.com/user-attachments/files/18208630/localization_config.zip)
2. Mount the extracted localization to the docker container e.g., `docker run -it -v $HOME/Downloads/localization_config:=/localization_config youtalk/autoware:universe-localization-mapping-20241220-amd64` 

3. Run the following commands:
```
ros2 launch tier4_localization_launch localization.launch.xml \
  pose_source:=ndt \
  twist_source:=twist_source \
  initial_pose:=[] \
  system_run_mode:=online \
  ndt_scan_matcher/pointcloud_preprocessor/crop_box_filter_measurement_range_param_path:=/localization_config/ndt_scan_matcher/pointcloud_preprocessor/crop_box_filter_measurement_range.param.yaml \
  ndt_scan_matcher/pointcloud_preprocessor/voxel_grid_downsample_filter_param_path:=/localization_config/ndt_scan_matcher/pointcloud_preprocessor/voxel_grid_filter.param.yaml \
  ndt_scan_matcher/pointcloud_preprocessor/random_downsample_filter_param_path:=/localization_config/ndt_scan_matcher/pointcloud_preprocessor/random_downsample_filter.param.yaml \
  ndt_scan_matcher/ndt_scan_matcher_param_path:=/localization_config/ndt_scan_matcher/ndt_scan_matcher.param.yaml \
  localization_error_monitor_param_path:=/localization_config/localization_error_monitor.param.yaml \
  ekf_localizer_param_path:=/localization_config/ekf_localizer.param.yaml \
  stop_filter_param_path:=/localization_config/stop_filter.param.yaml \
  pose_initializer_param_path:=/localization_config/pose_initializer.param.yaml \
  eagleye_param_path:=/localization_config/localization/eagleye_config.param.yaml \
  ar_tag_based_localizer_param_path:=/localization_config/ar_tag_based_localizer.param.yaml \
  lidar_marker_localizer/lidar_marker_localizer_param_path:=/localization_config/lidar_marker_localizer/lidar_marker_localizer.param.yaml \
  lidar_marker_localizer/pointcloud_preprocessor/crop_box_filter_measurement_range_param_path:=/localization_config/lidar_marker_localizer/pointcloud_preprocessor/crop_box_filter_measurement_range.param.yaml \
  lidar_marker_localizer/pointcloud_preprocessor/ring_filter_param_path:=/localization_config/lidar_marker_localizer/pointcloud_preprocessor/ring_filter.param.yaml \
  twist2accel_param_path:=/localization_config/twist2accel.param.yaml
```

If we are able to set default parameters for the launch files, then I think we could simplify the command, but that can be done as a separate PR.

Reply:
Thank you very much for your detail explanation. I understood. I will fix them.